src/csharp/Microsoft.Spark/Broadcast.cs

```diff
@@ -20,7 +20,6 @@ namespace Microsoft.Spark
     /// also attempts to distribute broadcast variables using efficient broadcast algorithms to
     /// reduce communication cost.
     /// </summary>
-    [Serializable]
     public sealed class Broadcast<T> : IJvmObjectReferenceProvider
```

Comment:
In order to register broadcast var in the registry on it being serialized, we can implement IMessagePackSerializationCallbackReceiver, and rename existing `OnSerialized` to `OnBeforeSerialize()`. Otherwise it's impossible to access var from worker context